pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -701,12 +701,7 @@ func (s *ServiceRuntime) putIntoObject(o client.Object, kon, resourceName string
 
 // GetObservedComposite returns the observed composite and unmarshals it into the given object.
 func (s *ServiceRuntime) GetObservedComposite(obj client.Object) error {
-	comp, err := request.GetObservedCompositeResource(s.req)
```

Comment:
This is a small optimization. During the initialization of the runtime, the observed composite is copied over to `observedComposite`. So no need to get it from the original request every time.